### ISSUE TITLE
Support Sparse Local Indexes

### ISSUE THREAD

**Issue by ajaishankar** [ServiceStack/PocoDynamo#2]
PutItem with a null value for an LocalIndex attribute throws a AmazonDynamoDBException - invalid attribute value type.

Sparse indexes with a missing sort key value is a very nice dynamodb feature:

http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GuidelinesForLSI.html#GuidelinesForLSI.SparseIndexes

```C#
class Poco
{
   public string HashKey { get; set; }
   public string RangeKey { get; set; }
   [Index]
   public string Bar { get; set; }
}

// throws exception
db.PutItem(new Poco() {
    HashKey = "Hash",
    RangeKey = "Range",
    Bar = null
});
```

**Comment by mythz** [comment]
Sparse Local Indexes should now be supported [from this commit](https://github.com/ServiceStack/ServiceStack.Aws/commit/5464589c0767fe4571c9fc22a8d6fd5010715258) where any properties with an `[Index]` or new `[ExcludeNullValue]` attribute will be skipped (i.e. have no attribute) instead of inserting a null value attribute.

```csharp
public class SparseLocalIndex
{
    public string HashKey { get; set; }
    public int RangeKey { get; set; }

    [Index]
    public string LocalIndex { get; set; }
}

db.PutItem(new SparseLocalIndex { HashKey = "A", RangeKey = 2, LocalIndex = null });

result = db.FromQuery<SparseLocalIndex>(x => x.HashKey == "A")
    .Filter(x => Dynamo.AttributeNotExists(x.LocalIndex))
    .Exec()
    .First();

Assert.That(result.RangeKey, Is.EqualTo(2));
```

There's also a global wide setting to skip **all** null values by default with:

```csharp
DynamoConfig.ExcludeNullValues = true;
```

This change is available from v4.0.51 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

Thanks for the feedback!

**Comment by ajaishankar** [comment]
Awesome thanks!
